tests/test_plots.py

```diff
@@ -49,3 +49,15 @@ def test_two_models_5_tries(self):
 
     with plt.ion():
       plot_benchmarks(misclass_rates, 'k', benchmark_range=[10, 25, 50, 100])
+
+  def test_modes(self):
+    misclass_rates = {
+      'Supervised Marker Map': np.array([[0.18801997, 0.23460899]]),
+      'Mixed Marker Map': np.array([[0.19966722, 0.26622296]])
+    }
+
+    with plt.ion():
+      plot_benchmarks(misclass_rates, 'k', mode='misclass', benchmark_range=[10,25])
+      plot_benchmarks(misclass_rates, 'k', mode='accuracy', benchmark_range=[10,25])
+      with pytest.raises(Exception):
+        plot_benchmarks(misclass_rates, 'k', mode='fake_mode', benchmark_range=[10,25])
```

Comment:
What does plt.ion do here? Is this with so that the plots are updated as these, possibly long running, benchmarks complete?

Reply:
The plot functions call `plt.show` which blocks further python execution until you close the little matplotlib windows that are generated. Since we are running a large battery of tests here, this means we would have to manually close the window for each call to plot_XXX. For these tests, I am just checking that the function runs without breaking (or breaks when it should, i.e. for mode="fake_mode"), rather than actually checking that the output plot looks correct.

I found the `plt.ion()` workaround, it makes it so that the `plt.show` is not blocking. So when I run the tests, python starts to generate the plots, but then the script ends and they are all closed.

Reply:
That makes a lot of sense. Thanks!